### ISSUE TITLE
JHV 4.0.0.10061fails to start on up-to-date Ubuntu 20.04

### ISSUE THREAD

**Issue by ebuchlin** [Helioviewer-Project/JHelioviewer-SWHV#142]
Maybe I wanted to try version 4 too early, in that case please disregard this issue...

I have downloaded `JHelioviewer_4_0_0_amd64.deb` and installed it with `dpkg -i`, on an up-to-date Ubuntu 20.04 (with the Plasma desktop on Xorg), and then run it from a terminal.

## Actual behavior

It fails to run, with an open dialog box displaying the last message in the terminal. The terminal displays:
```
2020-09-20 10:35:45,078 [main] INFO  root - JHelioviewer started with command-line options: 
2020-09-20 10:35:45,121 [main] INFO  root - JHV/SWHV-4.0.0.10061 (amd64 Linux 5.4.0-42-generic) AdoptOpenJDK JRE 11.0.8 7.8GB 8CPU
2020-09-20 10:35:46,453 [main] TRACE class org.ehcache.core.Ehcache-JPIPLevel - Initializing
2020-09-20 10:35:46,488 [main] DEBUG class org.ehcache.core.Ehcache-JPIPLevel - Initialize successful.
2020-09-20 10:35:46,511 [main] TRACE class org.ehcache.core.Ehcache-JPIPStream - Initializing
2020-09-20 10:35:46,533 [main] DEBUG class org.ehcache.core.Ehcache-JPIPStream - Initialize successful.
2020-09-20 10:35:47,040 [AWT-EventQueue-0] INFO  root - Start main window
2020-09-20 10:35:47,785 [AWT-EventQueue-0] ERROR root - GLInfo > Profile GL4bc is not available on X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x7fc6a4418120, owner true, ResourceToolkitLock[obj 0x4b00df9e, isOwner true, <2fb5dac6, 159bd420>[count 1, qsz 0, owner <AWT-EventQueue-0>]]], but: [GLProfile[GL4ES3/GL4.hw], GLProfile[GL2ES2/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GL3/GL4.hw], GLProfile[GL2GL3/GL4.hw]]
```
and then a few more log lines when I close the dialog box:
```
2020-09-20 10:35:53,780 [Thread-2] TRACE class org.ehcache.core.Ehcache-JPIPLevel - Closing
X11Util.Display: Shutdown (JVM shutdown: true, open (no close attempt): 1/1, reusable (open, marked uncloseable): 0, pending (open in creation order): 1)
X11Util: Open X11 Display Connections: 1
X11Util: Open[0]: NamedX11Display[:0, 0x7fc6a4418120, refCount 1, unCloseable false]
2020-09-20 10:35:53,880 [Thread-2] DEBUG class org.ehcache.core.Ehcache-JPIPLevel - Close successful.
2020-09-20 10:35:53,898 [Thread-2] TRACE class org.ehcache.core.Ehcache-JPIPStream - Closing
2020-09-20 10:35:53,906 [Thread-2] DEBUG class org.ehcache.core.Ehcache-JPIPStream - Close successful.
```

## Expected behavior

JHelioViewer starts

## Additional system information

From `glxgears -info` (I can also provide the output of `glxinfo`, which is much longer; it includes `direct rendering: Yes`):
```
GL_RENDERER   = Mesa Intel(R) UHD Graphics 620 (KBL GT2)
GL_VERSION    = 4.6 (Compatibility Profile) Mesa 20.0.8
GL_VENDOR     = Intel
GL_EXTENSIONS = GL_ARB_multisample GL_EXT_abgr GL_EXT_bgra GL_EXT_blend_color GL_EXT_blend_minmax GL_EXT_blend_subtract GL_EXT_copy_texture GL_EXT_subtexture GL_EXT_texture_object GL_EXT_vertex_array GL_EXT_compiled_vertex_array GL_EXT_texture GL_EXT_texture3D GL_IBM_rasterpos_clip GL_ARB_point_parameters GL_EXT_draw_range_elements GL_EXT_packed_pixels GL_EXT_point_parameters GL_EXT_rescale_normal GL_EXT_separate_specular_color GL_EXT_texture_edge_clamp GL_SGIS_generate_mipmap GL_SGIS_texture_border_clamp GL_SGIS_texture_edge_clamp GL_SGIS_texture_lod GL_ARB_framebuffer_sRGB GL_ARB_multitexture GL_EXT_framebuffer_sRGB GL_IBM_multimode_draw_arrays GL_IBM_texture_mirrored_repeat GL_3DFX_texture_compression_FXT1 GL_ARB_texture_cube_map GL_ARB_texture_env_add GL_ARB_transpose_matrix GL_EXT_blend_func_separate GL_EXT_fog_coord GL_EXT_multi_draw_arrays GL_EXT_secondary_color GL_EXT_texture_env_add GL_EXT_texture_filter_anisotropic GL_EXT_texture_lod_bias GL_INGR_blend_func_separate GL_NV_blend_square GL_NV_light_max_exponent GL_NV_texgen_reflection GL_NV_texture_env_combine4 GL_S3_s3tc GL_SUN_multi_draw_arrays GL_ARB_texture_border_clamp GL_ARB_texture_compression GL_EXT_framebuffer_object GL_EXT_texture_compression_s3tc GL_EXT_texture_env_combine GL_EXT_texture_env_dot3 GL_MESA_window_pos GL_NV_packed_depth_stencil GL_NV_texture_rectangle GL_ARB_depth_texture GL_ARB_occlusion_query GL_ARB_shadow GL_ARB_texture_env_combine GL_ARB_texture_env_crossbar GL_ARB_texture_env_dot3 GL_ARB_texture_mirrored_repeat GL_ARB_window_pos GL_ATI_fragment_shader GL_EXT_stencil_two_side GL_EXT_texture_cube_map GL_NV_depth_clamp GL_NV_fog_distance GL_APPLE_packed_pixels GL_ARB_draw_buffers GL_ARB_fragment_program GL_ARB_fragment_shader GL_ARB_shader_objects GL_ARB_vertex_program GL_ARB_vertex_shader GL_ATI_draw_buffers GL_ATI_texture_env_combine3 GL_ATI_texture_float GL_EXT_shadow_funcs GL_EXT_stencil_wrap GL_MESA_pack_invert GL_NV_primitive_restart GL_ARB_depth_clamp GL_ARB_fragment_program_shadow GL_ARB_half_float_pixel GL_ARB_occlusion_query2 GL_ARB_point_sprite GL_ARB_shading_language_100 GL_ARB_sync GL_ARB_texture_non_power_of_two GL_ARB_vertex_buffer_object GL_ATI_blend_equation_separate GL_EXT_blend_equation_separate GL_OES_read_format GL_ARB_color_buffer_float GL_ARB_pixel_buffer_object GL_ARB_texture_compression_rgtc GL_ARB_texture_float GL_ARB_texture_rectangle GL_EXT_packed_float GL_EXT_pixel_buffer_object GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_rectangle GL_EXT_texture_sRGB GL_EXT_texture_shared_exponent GL_ARB_framebuffer_object GL_EXT_framebuffer_blit GL_EXT_framebuffer_multisample GL_EXT_packed_depth_stencil GL_ARB_vertex_array_object GL_ATI_separate_stencil GL_EXT_draw_buffers2 GL_EXT_draw_instanced GL_EXT_gpu_program_parameters GL_EXT_gpu_shader4 GL_EXT_texture_array GL_EXT_texture_integer GL_EXT_texture_sRGB_decode GL_EXT_timer_query GL_OES_EGL_image GL_AMD_performance_monitor GL_EXT_texture_buffer_object GL_AMD_texture_texture4 GL_ARB_copy_buffer GL_ARB_depth_buffer_float GL_ARB_draw_instanced GL_ARB_half_float_vertex GL_ARB_instanced_arrays GL_ARB_map_buffer_range GL_ARB_texture_buffer_object GL_ARB_texture_rg GL_ARB_texture_swizzle GL_ARB_vertex_array_bgra GL_EXT_texture_swizzle GL_EXT_vertex_array_bgra GL_NV_conditional_render GL_AMD_conservative_depth GL_AMD_depth_clamp_separate GL_AMD_draw_buffers_blend GL_AMD_seamless_cubemap_per_texture GL_AMD_shader_stencil_export GL_ARB_ES2_compatibility GL_ARB_blend_func_extended GL_ARB_compatibility GL_ARB_debug_output GL_ARB_draw_buffers_blend GL_ARB_draw_elements_base_vertex GL_ARB_explicit_attrib_location GL_ARB_fragment_coord_conventions GL_ARB_provoking_vertex GL_ARB_sample_shading GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_ARB_shader_stencil_export GL_ARB_shader_texture_lod GL_ARB_tessellation_shader GL_ARB_texture_buffer_object_rgb32 GL_ARB_texture_cube_map_array GL_ARB_texture_gather GL_ARB_texture_multisample GL_ARB_texture_query_lod GL_ARB_texture_rgb10_a2ui GL_ARB_uniform_buffer_object GL_ARB_vertex_type_2_10_10_10_rev GL_EXT_provoking_vertex GL_EXT_texture_snorm GL_MESA_texture_signed_rgba GL_NV_texture_barrier GL_ARB_draw_indirect GL_ARB_get_program_binary GL_ARB_gpu_shader5 GL_ARB_gpu_shader_fp64 GL_ARB_robustness GL_ARB_separate_shader_objects GL_ARB_shader_bit_encoding GL_ARB_shader_precision GL_ARB_shader_subroutine GL_ARB_texture_compression_bptc GL_ARB_timer_query GL_ARB_transform_feedback2 GL_ARB_transform_feedback3 GL_ARB_vertex_attrib_64bit GL_ARB_viewport_array GL_EXT_direct_state_access GL_EXT_vertex_attrib_64bit GL_AMD_multi_draw_indirect GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_ARB_base_instance GL_ARB_compressed_texture_pixel_storage GL_ARB_conservative_depth GL_ARB_internalformat_query GL_ARB_map_buffer_alignment GL_ARB_shader_atomic_counters GL_ARB_shader_image_load_store GL_ARB_shading_language_420pack GL_ARB_shading_language_packing GL_ARB_texture_storage GL_ARB_transform_feedback_instanced GL_EXT_framebuffer_multisample_blit_scaled GL_EXT_transform_feedback GL_AMD_query_buffer_object GL_AMD_shader_trinary_minmax GL_AMD_vertex_shader_layer GL_AMD_vertex_shader_viewport_index GL_ARB_ES3_compatibility GL_ARB_arrays_of_arrays GL_ARB_clear_buffer_object GL_ARB_compute_shader GL_ARB_copy_image GL_ARB_explicit_uniform_location GL_ARB_fragment_layer_viewport GL_ARB_framebuffer_no_attachments GL_ARB_invalidate_subdata GL_ARB_multi_draw_indirect GL_ARB_program_interface_query GL_ARB_robust_buffer_access_behavior GL_ARB_shader_image_size GL_ARB_shader_storage_buffer_object GL_ARB_stencil_texturing GL_ARB_texture_buffer_range GL_ARB_texture_query_levels GL_ARB_texture_storage_multisample GL_ARB_texture_view GL_ARB_vertex_attrib_binding GL_KHR_debug GL_KHR_robustness GL_KHR_texture_compression_astc_ldr GL_AMD_pinned_memory GL_ARB_buffer_storage GL_ARB_clear_texture GL_ARB_enhanced_layouts GL_ARB_indirect_parameters GL_ARB_internalformat_query2 GL_ARB_multi_bind GL_ARB_query_buffer_object GL_ARB_seamless_cubemap_per_texture GL_ARB_shader_draw_parameters GL_ARB_shader_group_vote GL_ARB_shading_language_include GL_ARB_texture_mirror_clamp_to_edge GL_ARB_texture_stencil8 GL_ARB_vertex_type_10f_11f_11f_rev GL_EXT_shader_framebuffer_fetch GL_EXT_shader_integer_mix GL_INTEL_performance_query GL_ARB_ES3_1_compatibility GL_ARB_clip_control GL_ARB_conditional_render_inverted GL_ARB_cull_distance GL_ARB_derivative_control GL_ARB_direct_state_access GL_ARB_get_texture_sub_image GL_ARB_pipeline_statistics_query GL_ARB_shader_texture_image_samples GL_ARB_texture_barrier GL_ARB_transform_feedback_overflow_query GL_EXT_polygon_offset_clamp GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_KHR_context_flush_control GL_KHR_robust_buffer_access_behavior GL_ARB_fragment_shader_interlock GL_ARB_gpu_shader_int64 GL_ARB_parallel_shader_compile GL_ARB_post_depth_coverage GL_ARB_shader_atomic_counter_ops GL_ARB_shader_ballot GL_ARB_shader_clock GL_ARB_shader_viewport_layer_array GL_EXT_shader_samples_identical GL_KHR_no_error GL_KHR_texture_compression_astc_sliced_3d GL_NV_fragment_shader_interlock GL_ARB_gl_spirv GL_ARB_spirv_extensions GL_MESA_shader_integer_functions GL_ARB_polygon_offset_clamp GL_ARB_texture_filter_anisotropic GL_KHR_parallel_shader_compile GL_EXT_EGL_image_storage GL_EXT_shader_framebuffer_fetch_non_coherent GL_EXT_texture_sRGB_R8 GL_EXT_texture_shadow_lod GL_INTEL_shader_atomic_float_minmax GL_INTEL_shader_integer_functions2 GL_MESA_framebuffer_flip_y GL_NV_compute_shader_derivatives GL_EXT_EGL_sync GL_EXT_demote_to_helper_invocation 
VisualID 1090, 0x442
```
`java -version`:
```
openjdk version "14.0.1" 2020-04-14
OpenJDK Runtime Environment (build 14.0.1+7-Ubuntu-1ubuntu1)
OpenJDK 64-Bit Server VM (build 14.0.1+7-Ubuntu-1ubuntu1, mixed mode, sharing)
```
`update-alternatives --query java`:
```
Name: java
Link: /usr/bin/java
Slaves:
 java.1.gz /usr/share/man/man1/java.1.gz
Status: auto
Best: /usr/lib/jvm/java-14-openjdk-amd64/bin/java
Value: /usr/lib/jvm/java-14-openjdk-amd64/bin/java

Alternative: /usr/lib/jvm/java-11-openjdk-amd64/bin/java
Priority: 1111
Slaves:
 java.1.gz /usr/lib/jvm/java-11-openjdk-amd64/man/man1/java.1.gz

Alternative: /usr/lib/jvm/java-14-openjdk-amd64/bin/java
Priority: 1411
Slaves:
 java.1.gz /usr/lib/jvm/java-14-openjdk-amd64/man/man1/java.1.gz
```
Note: OpenJDK 11 seems to be used by JHelioViewer as it seems to be "recommended" (whatever this means) for AMD64 (see description of the `default-jre` Ubuntu package) although the chosen alternative (see above) is OpenJDK 14...

**Comment by bogdanni** [comment]
Thanks for the great report!
Linux packages bundle now a JRE to reduce the potential problems on users' computers, so it is not using the system Java.

This looks like a case of JOGL library mis-detection of the video card. It works on my Ubuntu virtual machine. I will try to investigate, but this is very difficult.

**Comment by bogdanni** [comment]
Could you please retry with the new Debian package from the test directory? It should identify itself as 4.0.0.10062.
Both worked in a Kubuntu virtual machine, so I still think it's an issue JOGL getting confused by the OpenGL level capabilities returned by some drivers.

**Comment by ebuchlin** [comment]
Different player, try again :-)
```
2020-09-20 13:35:21,352 [main] INFO  root - JHelioviewer started with command-line options: 
2020-09-20 13:35:21,393 [main] INFO  root - JHV/SWHV-4.0.0.10062 (amd64 Linux 5.4.0-42-generic) AdoptOpenJDK JRE 11.0.8 7.8GB 8CPU
2020-09-20 13:35:22,332 [main] TRACE class org.ehcache.core.Ehcache-JPIPLevel - Initializing
2020-09-20 13:35:22,525 [main] DEBUG class org.ehcache.core.Ehcache-JPIPLevel - Initialize successful.
2020-09-20 13:35:22,680 [main] TRACE class org.ehcache.core.Ehcache-JPIPStream - Initializing
2020-09-20 13:35:22,711 [main] DEBUG class org.ehcache.core.Ehcache-JPIPStream - Initialize successful.
2020-09-20 13:35:23,650 [AWT-EventQueue-0] INFO  root - Start main window
2020-09-20 13:35:24,570 [AWT-EventQueue-0] ERROR root - GLInfo > Profile GL4bc is not available on X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x7f00e0559e50, owner true, ResourceToolkitLock[obj 0x2e196854, isOwner true, <77b95501, 24c5a3c3>[count 1, qsz 0, owner <AWT-EventQueue-0>]]], but: [GLProfile[GL4ES3/GL4.hw], GLProfile[GL2ES2/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GL3/GL4.hw], GLProfile[GL2GL3/GL4.hw]]
```
(failure dialog box, then: )
```
2020-09-20 13:35:32,800 [Thread-2] TRACE class org.ehcache.core.Ehcache-JPIPLevel - Closing
2020-09-20 13:35:32,826 [Thread-2] DEBUG class org.ehcache.core.Ehcache-JPIPLevel - Close successful.
X11Util.Display: Shutdown (JVM shutdown: true, open (no close attempt): 1/1, reusable (open, marked uncloseable): 0, pending (open in creation order): 1)
X11Util: Open X11 Display Connections: 1
X11Util: Open[0]: NamedX11Display[:0, 0x7f00e0559e50, refCount 1, unCloseable false]
2020-09-20 13:35:32,850 [Thread-2] TRACE class org.ehcache.core.Ehcache-JPIPStream - Closing
2020-09-20 13:35:32,861 [Thread-2] DEBUG class org.ehcache.core.Ehcache-JPIPStream - Close successful.
```
I understand this must be difficult to investigate...
I don't have the issue on a Debian 10 (on a 12-yr-old PC...).

**Comment by bogdanni** [comment]
Could you please try again? Thanks!

**Comment by ebuchlin** [comment]
Still the same...
```
2020-09-22 23:41:43,332 [main] INFO  root - JHelioviewer started with command-line options: 
2020-09-22 23:41:43,393 [main] INFO  root - JHV/SWHV-4.0.0.10063 (amd64 Linux 5.4.0-42-generic) AdoptOpenJDK JRE 11.0.8 7.8GB 8CPU
2020-09-22 23:41:44,677 [main] TRACE class org.ehcache.core.Ehcache-JPIPLevel - Initializing
2020-09-22 23:41:44,713 [main] DEBUG class org.ehcache.core.Ehcache-JPIPLevel - Initialize successful.
2020-09-22 23:41:44,740 [main] TRACE class org.ehcache.core.Ehcache-JPIPStream - Initializing
2020-09-22 23:41:44,764 [main] DEBUG class org.ehcache.core.Ehcache-JPIPStream - Initialize successful.
2020-09-22 23:41:45,479 [AWT-EventQueue-0] INFO  root - Start main window
2020-09-22 23:41:46,238 [AWT-EventQueue-0] ERROR root - GLInfo > Profile GL4bc is not available on X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x7f5b26a3ffb0, owner true, ResourceToolkitLock[obj 0x106ee18c, isOwner true, <780663aa, 70140762>[count 1, qsz 0, owner <AWT-EventQueue-0>]]], but: [GLProfile[GL4ES3/GL4.hw], GLProfile[GL2ES2/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GL3/GL4.hw], GLProfile[GL2GL3/GL4.hw]]
2020-09-22 23:41:49,690 [Thread-2] TRACE class org.ehcache.core.Ehcache-JPIPLevel - Closing
X11Util.Display: Shutdown (JVM shutdown: true, open (no close attempt): 1/1, reusable (open, marked uncloseable): 0, pending (open in creation order): 1)
X11Util: Open X11 Display Connections: 1
X11Util: Open[0]: NamedX11Display[:0, 0x7f5b26a3ffb0, refCount 1, unCloseable false]
2020-09-22 23:41:49,782 [Thread-2] DEBUG class org.ehcache.core.Ehcache-JPIPLevel - Close successful.
2020-09-22 23:41:49,799 [Thread-2] TRACE class org.ehcache.core.Ehcache-JPIPStream - Closing
2020-09-22 23:41:49,807 [Thread-2] DEBUG class org.ehcache.core.Ehcache-JPIPStream - Close successful.
```

**Comment by bogdanni** [comment]
I assume this was fixed together with #139. If not, please reopen.

**Comment by ebuchlin** [comment]
Now it works, thanks!
```
2020-09-23 17:24:17,068 [main] INFO  root - JHelioviewer started with command-line options: 
2020-09-23 17:24:17,110 [main] INFO  root - JHV/SWHV-4.0.0.10069 (amd64 Linux 5.4.0-42-generic) AdoptOpenJDK JRE 11.0.8 7.8GB 8CPU
2020-09-23 17:24:17,849 [main] TRACE class org.ehcache.core.Ehcache-JPIPLevel - Initializing
2020-09-23 17:24:17,888 [main] DEBUG class org.ehcache.core.Ehcache-JPIPLevel - Initialize successful.
2020-09-23 17:24:17,913 [main] TRACE class org.ehcache.core.Ehcache-JPIPStream - Initializing
2020-09-23 17:24:17,943 [main] DEBUG class org.ehcache.core.Ehcache-JPIPStream - Initialize successful.
2020-09-23 17:24:18,530 [AWT-EventQueue-0] INFO  root - Start main window
2020-09-23 17:24:19,979 [AWT-EventQueue-0] INFO  root - Load bundled plugins
2020-09-23 17:24:20,551 [AWT-EventQueue-0] INFO  root - GLInfo > OpenGL 4.6 (Core Profile) Mesa 20.0.8
2020-09-23 17:24:28,207 [Worker] DEBUG root - Response: {"frames":97,"message":null,"uri":"jpip://helioviewer.ias.u-psud.fr:8080/movies/SDO_AIA_171_F2020-09-21T14.44.09Z_T2020-09-23T14.44.09ZB1800L.jpx"}
```
